src/mlagility/api/cloud.py

```diff
@@ -437,3 +555,94 @@ def execute_cpu_remotely(
             )
     # Stop redirecting stdout
     sys.stdout = sys.stdout.terminal
+
+
+def execute_cpu_locally(
+    state: build.State, log_execute_path: str, iterations: int
+) -> None:
+    """
+    Execute Model on the local CPU
+    """
+
+    # Redirect all stdout to log_file
+    sys.stdout = build.Logger(log_execute_path)
+
+    # Setup local execution folders to save outputs/ errors
+    username = getpass.getuser()
+    output_dir = f"/home/{username}/mlagility_local_cache"
+    outputs_file = f"{output_dir}/outputs.txt"
+    errors_file = f"{output_dir}/errors.txt"
+
+    setup_local_host(device_type="cpu", output_dir=output_dir)
+
+    # Check if ONNX file has been generated
+    if not os.path.exists(state.converted_onnx_file):
+        msg = "Model file not found"
+        raise exp.GroqModelRuntimeError(msg)
+
+    os.makedirs(f"{output_dir}/onnxmodel")
+    shutil.copy(state.converted_onnx_file, f"{output_dir}/onnxmodel/model.onnx")
+
+    # Check if conda is installed
+    conda_location = shutil.which("conda")
+    if not conda_location:
+        raise ValueError("conda installation not found.")
+    conda_src = conda_location.split("miniconda3")[0]
+
+    # Create/ update local conda environment for CPU benchmarking
+    print("Creating environment...")
+    env_name = "onnxruntime-env"
```

Comment:
```suggestion
    env_name = "mlagility-onnxruntime-env"
```

namespace it to avoid collisions with the user's envs